freud/box.pyx

```diff
@@ -736,6 +728,20 @@ cdef class Box:
         else:
             raise ValueError("Box can only be multiplied by positive values.")
 
+    def __mul__(arg1, arg2):
+        # NOTE: cython<3.0 treats __mul__ and __rmul__ as one operation, so
+        # type checks are needed. For cython>=3.0, __rmul__ is implemented
```

Comment:
Do you need to support both Cython < 3 and Cython >= 3? I would strongly recommend requiring Cython 3, which would mean this note isn't needed.

Reply:
Why require Cython >= 3? Won't people want to be able to build freud with older cython versions too?

Reply:
I don't think this will be a problem. I think you can require Cython 3 with no harm done. Most users will only need prebuilt packages, and those who do wish to build from source can do so with build isolation (on by default), which will install Cython into a temporary environment. Most Cython-using packages I know are just updating the requirement to 3.0.

Reply:
This was marked as resolved but does not appear to be resolved. I expected the comment would be deleted, and this would not require a utility method `_mul_impl`. Try just implementing `__mul__` and call that in `__rmul__`.

Reply:
Should be good now.